src/amo/reducers/users.js

```diff
@@ -0,0 +1,256 @@
+/* @flow */
+import { ALL_SUPER_POWERS } from 'core/constants';
+
+
+export const LOG_OUT_USER: 'LOG_OUT_USER' = 'LOG_OUT_USER';
+export const LOAD_CURRENT_USER_ACCOUNT: 'LOAD_CURRENT_USER_ACCOUNT' = 'LOAD_CURRENT_USER_ACCOUNT';
+export const FETCH_CURRENT_USER_ACCOUNT: 'FETCH_CURRENT_USER_ACCOUNT' = 'FETCH_CURRENT_USER_ACCOUNT';
+export const FETCH_USER_ACCOUNT: 'FETCH_USER_ACCOUNT' = 'FETCH_USER_ACCOUNT';
+export const LOAD_USER_ACCOUNT: 'LOAD_USER_ACCOUNT' = 'LOAD_USER_ACCOUNT';
+
+export type UserId = string;
+
+export type UserType = {|
+  // Basic user account object fields, returned by the API.
+  average_addon_rating: number,
+  biography: ?string,
+  created: ?string,
+  has_anonymous_display_name: boolean,
+  has_anonymous_username: boolean,
+  homepage: string | null,
+  id: ?number,
+  is_addon_developer: boolean,
+  is_artist: boolean,
+  location: string | null,
+  name: string,
+  num_addons_listed: number,
+  occupation: string | null,
+  picture_type: string | null,
+  picture_url: string | null,
+  username: ?string,
+  // Properties returned if we are accessing our own profile or the current user
+  // has the `Users:Edit` permission.
+  deleted?: boolean,
+  display_name?: string | null,
+  email?: string,
+  last_login?: string,
+  last_login_ip?: string,
+  is_verified?: boolean,
+  permissions: ?Array<string>,
+  read_dev_agreement?: boolean,
+  // Properties we add to each object.
+  displayName: ?string,
+|};
+
+export type UsersStateType = {
```

Comment:
I don't see anything to indicate that user data is loading. If you think it's fine to add later then it can be postponed. I guess you could add it later like:

```js
export type UsersStateType = {
  currentUserID: number | null,
  byID: { [userId: number]: UserType },
  byUsername: { [username: string]: UserId },
  developerProfilePage: {
    loading: boolean,
    user: UserId | null,
  },
};
```

Theoretically, `currentUser` should have a way to indicate loading state but since that will always happen server side, I'm not worried about it. It would be a lot of work to change that!

Reply:
I was going to add it later in with the actual pages but no harm in doing so now. 😄